packages/reva-tests/cypress/e2e/trainingProgram.cy.js

```diff
@@ -0,0 +1,51 @@
+import { stubQuery } from "../utils/graphql";
+
+context("Training Program", () => {
+    describe('Testing Descriptions', () => {
+        it("display all fields", () => {
+            cy.intercept("POST", "/graphql", (req) => {
+                stubQuery(req, "getCandidacy", "getCandidacy3.json");
+            });
+            cy.visit("/");
+            cy.wait("@getCandidacy");
+            cy.get('dl').should("exist").should('have.length', 1)
+            cy.get('dl > dt').should("exist").should('have.length', 8)
+            cy.get('dl > dd').should("exist").should('have.length', 8)
+
+        });
+
+        it("don't display missing fields", () => {
+            cy.intercept("POST", "/graphql", (req) => {
+                stubQuery(req, "getCandidacy", "getCandidacy3_missingFields.json");
+            });
+            cy.visit("/");
+            cy.wait("@getCandidacy");
+            cy.get('dl').should("exist").should('have.length', 1)
+            cy.get('dl > dt').should("exist").should('have.length', 4)
+            cy.get('dl > dd').should("exist").should('have.length', 4)
+
+        });
+    })
+
+
+    describe('Testing Checkbox logic', () => {
+        it("validates checked condition and its mechanics", () => {
+            cy.intercept("POST", "/graphql", (req) => {
+                stubQuery(req, "getCandidacy", "getCandidacy3.json");
+            });
+            cy.visit("/");
+            cy.wait("@getCandidacy");
+
+            const checkbox = cy.get('[data-test="checkbox-accept-conditions"]');
```

Comment:
Les commandes Cypress sont schedulées et lancées de façon asynchrone. Donc pour éviter tout soucis on va priviligier :
- le travail dans une closure (avec `.then`) 
- ou bien la création d'un alias (avec `.as`)

Reply:
oui mais là je n'utilise que des méthodes cypress, elles devraient etre exécutés dans l'ordre non ?

Reply:
[doc cypress](https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Return-Values)

Reply:
utilisation d'[aliases](https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Aliases)